### PR TITLE
Try vm-actions instead of cross-platform-actions

### PR DESCRIPTION
**Pull request by davidchisnall**
This uses VirtualBox instead of xhyve.  It might be slower, but should be more reliable.

### CODE REVIEW

**Comment by davidchisnall** [comment]
This is running the tests on {Net,Free,Open}BSD.  The tests currently fail on Net- and OpenBSD, though for different reasons.  I've wired up FreeBSD to the all-checks action, but not the other two, so the failing tests don't break merging but it's easy to see if a commit fixes them (once they're fixed, hopefully we can keep them from breaking again).

**Comment by davidchisnall** [comment]
@devnexen, it looks as if some of the tests fail on NetBSD as well.

File: CMakeLists.txt [hunk 1]
```diff
@@ -288,7 +288,7 @@ if(NOT SNMALLOC_HEADER_ONLY_LIBRARY)
 
   if(NOT (DEFINED SNMALLOC_LINKER_FLAVOUR) OR ("${SNMALLOC_LINKER_FLAVOUR}" MATCHES "^$"))
     # Linker not specified externally; probe to see if we can make lld work
-    set(CMAKE_REQUIRED_LINK_OPTIONS -fuse-ld=lld)
+    set(CMAKE_REQUIRED_LINK_OPTIONS -fuse-ld=lld -Wl,--icf=all)
```
Comment (mjp41) [inline CMakeLists.txt:291]: Why did you remove this?  It makes the library smaller as there are a lot of names for the same function in the API surface (e.g. lots of new overloads and malloc can all be folded into a single thing).

Comment (mjp41) [reply CMakeLists.txt:291]: FYI: this change increases the size of `libsnmallocshim.so` by about 10%.

**Comment by devnexen** [comment]
> @devnexen, it looks as if some of the tests fail on NetBSD as well.

it comes down to the access enforcing thing. Once disabled all pass.

File: CMakeLists.txt (hunk 1 above)
Comment (davidchisnall) [reply CMakeLists.txt:291]: I didn't remove it, I added it to the check that enables it.  Without this, we now have CI that triggers the reason that I have to carry a local patch for snmalloc to disable this: If you compile with GCC, the `-fuse-ld=lld` test fails (GCC consumes but silently ignores the flag?  Or just warns and doesn't error?), but then GCC passes this flag to ld.bfd and it fails to link.

**Comment by davidchisnall** [comment]
@devnexen, please feel free to push to this branch or raise a PR against it.

File: CMakeLists.txt (hunk 1 above)
Comment (mjp41) [reply CMakeLists.txt:291]: Thanks. I miss read the diff.

**Comment by mjp41** [comment]
Can we merge this to unblock the CI?
